tests/suites/test_suite_entropy.function

```diff
@@ -565,3 +566,19 @@ exit:
     MD_PSA_DONE();
 }
 /* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_PLATFORM_GET_ENTROPY_ALT */
+void entropy_test_platform_get_entropy_alt(void)
+{
+    uint8_t data[MBEDTLS_ENTROPY_POLL_PLATFORM_MIN];
+    size_t olen;
+
+    TEST_ASSERT(mbedtls_entropy_poll_platform(NULL, data, sizeof(data), &olen) == 0);
```

Comment:
Do we have a test somewhere (presumably in `test_suite_entropy.function`) to validate what happens if the callback reports full entropy but with a smaller output size, i.e. `*olen < output_size`? There's a loop in `entropy.c` that should call the function again until it provides enough entropy, but I'm not sure our tests are up to date with the change we've made to the callback: we'd need to let the test callback know to report less entropy, and I don't see this being done.